lib/src/components/NxToast/NxToastProvider.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+
+import React, { useCallback, useEffect, useState } from 'react';
+import ToastContext from './contexts';
+import NxToastContainer from './NxToastContainer';
+import { NxToastProviderProps, nxToastProviderPropTypes, ToastAddModel, ToastModel } from './types';
+
+let id = 1;
+
+const NxToastProvider = (props: NxToastProviderProps) => {
+  const { children } = props;
+  const [toasts, setToasts] = useState<ToastModel[]>([]);
+  const [activeElementBeforeToast, setActiveElementBeforeToast] = useState<HTMLElement | null>(null);
+
+  const addToast = useCallback((content: ToastAddModel) => {
+    //Before displaying the toast, keep track of active element
+    const currentFocusedElement = document.activeElement as HTMLElement;
+
+    //When the toast is displayed, the focus is on the Toast's close button
+    //In this case, the active element is the Toast's close button, so ignore it
+    if (!currentFocusedElement.classList.contains('nx-toast__close')) {
+      setActiveElementBeforeToast(currentFocusedElement);
+    }
+
+    setToasts((toasts) => [
+      ...toasts,
+      { toastId: id++, type: content.type, message: content.message }
+    ]);
+  }, [setToasts]);
+
+  const removeToast = useCallback((id: number) => {
+    setToasts(toasts => toasts.filter(t => t.toastId !== id));
```

Comment:
```suggestion
    setToasts(filter(propEq('toastId', id)));
```

Reply:
When trying to implement filter(propEq('toastId', id)));, we found that the function was removing all the toasts EXCEPT the toast that was clicked on to remove. So we found this as an alternative:

Reply:
When trying to implement filter(propEq('toastId', id)));, we found that the function was removing all the toasts EXCEPT the toast that was clicked on to remove. So we found this as an alternative:

Reply:
When trying to implement filter(propEq('toastId', id)));, we found that the function was removing all the toasts EXCEPT the toast that was clicked on to remove. So we found this as an alternative:  "toasts.filter(complement(propEq('toastId', id)));" .  However, we are unsure if this is any cleaner than what Abhinav already had.

Reply:
> > When trying to implement filter(propEq('toastId', id)));, we found that the function was removing all the toasts EXCEPT the toast that was clicked on to remove. So we found this as an alternative:
> > "toasts.filter(complement(propEq('toastId', id)));" .
> > However, we are unsure if this is any cleaner than what Abhinav already had.
> 
> My bad, what we're looking for is [`reject`](https://ramdajs.com/docs/#reject)

Thanks for the clarification. Implemented with 'reject' instead of 'filter' here: [ef7f5e3](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/ef7f5e312e61438d9614fe21256e223a6ce00df2)